package/MDAnalysis/analysis/parallel.py

```diff
@@ -0,0 +1,375 @@
+            and self.n_workers > 0: f"n_workers should be positive integer, got {self.n_workers=}",
+        }
+
+    def _get_warnings(self):
+        """Get dictionary with `condition: warning_message` pairs that ensure the good usage of the backend instance
+
+        Returns
+        -------
+        dict
+            dictionary with `condition: warning_message` pairs that will get checked during _validate() run
+
+        .. versionadded: 2.7.0
+        """
+        return dict()
+
+    def _validate(self):
+        """Check correctness (e.g. `dask` is installed if using `backend='dask'`)
+        and good usage (e.g. `n_workers=1` if backend is serial) of the backend
+
+        Raises
+        ------
+        ValueError
+            if one of the conditions in :meth:`self._get_checks()` is True
+
+        .. versionadded: 2.7.0
+        """
+        for check, msg in self._get_checks().items():
+            if not check:
+                raise ValueError(msg)
+        for check, msg in self._get_warnings().items():
+            if not check:
+                warnings.warn(msg)
+
+    def apply(self, func: Callable, computations: list) -> list:
+        """Main function that will get called when using an instance of an object, mapping function to all tasks
+        in the `computations` list. Should effectively be equivalent to running [func(item) for item in computations]
+        while using the parallel backend capabilities.
+
+        Parameters
+        ----------
+        func : Callable
+            function to be called on each of the tasks in computations list
+        computations : list
+            computation tasks to apply function to
+
+        Returns
+        -------
+        list
+            list of results of the function
+
+        .. versionadded: 2.7.0
+        """
+        raise NotImplementedError("Should be re-implemented in subclasses")
+
+
+class BackendSerial(BackendBase):
+    """A built-in backend that does serial execution of the function, without any parallelization
+
+    .. versionadded: 2.7.0
+    """
+
+    def _get_warnigns(self):
+        return {self.n_workers > 1, "n_workers > 1 will be ignored while executing with backend='serial'"}
+
+    def apply(self, func: Callable, computations: list) -> list:
+        return [func(task) for task in computations]
+
+
+class BackendMultiprocessing(BackendBase):
+    """A built-in backend that executes a given function using multiprocessing.Pool.map method
+
+    .. versionadded: 2.7.0
+    """
+
+    def apply(self, func: Callable, computations: list) -> list:
+        from multiprocessing import Pool
+
+        with Pool(processes=self.n_workers) as pool:
+            results = pool.map(func, computations)
+        return results
+
+
+class BackendDask(BackendBase):
+    """A built-in backend that executes a given function using dask.delayed.compute method with `scheduler='processes'`
+    and `chunksize=1`. Requires `dask` module to be installed.
```

Comment:
I'd expand the documentation a bit to explain what `scheduler='process'` and `chunksize=1` do.